src/poly/poly.jl

```diff
@@ -278,7 +278,7 @@ end
 ###############################################################################
 
 function Base.hash(p::spoly{T}, h::UInt) where T <: Nemo.RingElem
-   return h # FIXME: Implement a useful hash function instead
+   return total_degree(p) * h # FIXME: Implement a more useful hash function instead
```

Comment:
since `Base.hash` by default sets `h=UInt(0)`, it's not really much better than just `h` ;-)

I have no idea what are the inners of `spoly`, but a simple implementation could be
`hash(p::spoly{T}, h::UInt) where T = xor(hash(spoly), hash(coefficients_or_something_unique_to(p), h))`
`hash(spoly)` can be of course precomputed first (add a comment). 
Or if You can iterate over coefficients of `p` then

`hash(p::spoly, h::UInt) = xor(HASH_SPOLY, reduce(xor, hash(c, h) for c in coeffs(p))`

But I'd really like to see it as `UInt(0)` globally.

Reply:
You can't make the hash depend on the object pointer at all. You have to use the data in the actual polynomial, i.e. the coefficients.

For about the fifth time, this is not difficult to do in the slightest and we have examples of how to do it in Nemo and AbstractAlgebra.

Reply:
@kalmarek I hope the version in 2a927c5e6c8f21ddd550e9773c0a131fbeffafe8 is a little better.